python/pyspark/sql/dataframe.py

```diff
@@ -451,10 +451,10 @@ def repartition(self, numPartitions, *cols):
         +---+-----+
         |age| name|
         +---+-----+
-        |  5|  Bob|
-        |  5|  Bob|
         |  2|Alice|
+        |  5|  Bob|
         |  2|Alice|
+        |  5|  Bob|
         +---+-----+
```

Comment:
I'm confused here: why the results differ from each other when the top level re-partitioning condition stays the same?